core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -426,7 +420,7 @@ public Version getVersion() {
         }
 
         public List<Channel> getChannels() {
-            return Arrays.asList(channels);
+            return new ArrayList<>(channels);
```

Comment:
Can we wrap this in the ctor in a `Collections#unmodifiableList` and just return the instance instead of a mutable copy

Reply:
Yep good idea. I should use the unmodifiableList in a few other places (in NodeChannels) too.